vars/pullGitRepositoryToSapCloudPlatformAbapEnvironment.groovy

```diff
@@ -0,0 +1,185 @@
+import com.sap.piper.Utils
+import groovy.json.JsonSlurper
+import hudson.AbortException
+import groovy.transform.Field
+import org.jenkinsci.plugins.workflow.steps.FlowInterruptedException
+
+@Field def STEP_NAME = getClass().getName()
+@Field Set GENERAL_CONFIG_KEYS = [
+    /**
+     * Specifies the host address of the SAP Cloud Platform ABAP Environment system
+     */
+    'host',
+    /**
+     * Specifies the name of the Repository (Software Component) on the SAP Cloud Platform ABAP Environment system
+     */
+    'repositoryName'
+]
+@Field Set STEP_CONFIG_KEYS = GENERAL_CONFIG_KEYS.plus([
+    /**
+     * Specifies the communication user of the communication scenario SAP_COM_0510
+     */
+    'username',
+    /**
+     * Specifies the password of the communication user
+     */
+    'password'
+])
+@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
+/**
+ * Pulls a Repository (Software Component) to a SAP Cloud Platform ABAP Environment system.
+ *
+ * !!! note "Git Repository and Software Component"
+ *       In SAP Cloud Platform ABAP Environment Git repositories are wrapped in Software Components (which are managed in the App "Manage Software Components")
+ *       Currently, those two names are used synonymous.
+ */
+@GenerateDocumentation
+void call(Map parameters = [:]) {
+
+    handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters, failOnError: true) {
+
+        def script = checkScript(this, parameters) ?: this
+
+        ConfigurationHelper configHelper = ConfigurationHelper.newInstance(this)
+            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
+            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS)
+            .mixin(parameters, PARAMETER_KEYS)
+
+        Map configuration = configHelper.use()
+
+        configHelper
+            .withMandatoryProperty('host', 'Host not provided')
+            .withMandatoryProperty('repositoryName', 'Repository / Software Component not provided')
+            .withMandatoryProperty('username')
+            .withMandatoryProperty('password')
+
+        String usernameColonPassword = configuration.username + ":" + configuration.password
+        String authToken = usernameColonPassword.bytes.encodeBase64().toString()
+        String urlString = configuration.host + '/sap/opu/odata/sap/MANAGE_GIT_REPOSITORY/Pull'
+        echo "[${STEP_NAME}] General Parameters: URL = \"${urlString}\", repositoryName = \"${configuration.repositoryName}\""
+
+        String urlPullEntity = triggerPull(configuration, urlString, authToken)
+
+        if (urlPullEntity != null) {
+            String finalStatus = pollPullStatus(urlPullEntity, authToken)
+            if (finalStatus != 'S') {
+                error "[${STEP_NAME}] Pull Failed"
+            }
+        }
+    }
+}
+
+private String triggerPull(Map configuration, String url, String authToken) {
+
+    String entityUri = null
+    String headerFile = "headerAuth.txt"
```

Comment:
I guess we should delete the headerFile(s) in some kind of cleanup action in order not to expose the cookie(s). Most probably they are invalid after a logoff (... is there any logoff), but in order to not provide an attack vector we should delete the file afterwards anyway.